### PR TITLE
[16.0] stock_release_channel: multi-company support

### PR DESCRIPTION
**Pull request by sebalix**

### CODE REVIEW

File: stock_release_channel/models/stock_picking.py [hunk 1]
```diff
@@ -84,6 +84,7 @@ def _get_release_channel_possible_candidate_domain(self):
         self.ensure_one()
         return [
             ("state", "!=", "asleep"),
+            ("company_id", "=", self.picking_type_id.company_id.id),
```
Comment (jbaudoux) [inline stock_release_channel/models/stock_picking.py]:
```suggestion
            ("company_id", "=", self.company_id.id),
```

File: stock_release_channel/models/stock_release_channel.py [hunk 2]
```diff
@@ -52,6 +60,7 @@ class StockReleaseChannel(models.Model):
         domain="warehouse_id"
         " and [('warehouse_id', '=', warehouse_id), ('code', '=', 'outgoing')]"
         " or [('code', '=', 'outgoing')]",
+        check_company=True,
```
Comment (jbaudoux) [inline stock_release_channel/models/stock_release_channel.py]: You cannot use both domain and check_company on the field. You need to add `("company_id", "in", (company_id, False))` in the domain

File: stock_release_channel/models/stock_release_channel.py [hunk 3]
```diff
@@ -209,6 +218,7 @@ class StockReleaseChannel(models.Model):
         column2="partner_id",
         string="Partners",
         context={"active_test": False},
+        domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]",
```
Comment (jbaudoux) [inline stock_release_channel/models/stock_release_channel.py]: Here you can use
```suggestion
        check_company=True,
```

**Review by jbaudoux — CHANGES_REQUESTED** [review]
I would also add on:
* stock_release_channel
  * picking_ids `check_company=True` 
* shipment_advice
  * dock_id : `domain="[('warehouse_id', '=', warehouse_id)]"`
  * planned_move_ids `check_company=True`
  * loaded_move_lines_ids `check_company=True`
  * loaded_move_lines_without_package_ids `check_company=True`

File: stock_release_channel/models/stock_picking.py (hunk 1 above)
Comment (sebalix) [reply stock_release_channel/models/stock_picking.py]: `<stock.picking>.company_id` is not required, while it is on `<stock.picking.type>`

Comment (jbaudoux) [reply stock_release_channel/models/stock_picking.py]: But you are processing a picking, so better use its company.
I guess they forgot the required=True and we can assume there will always be one

**Review by jbaudoux — APPROVED** [review]
LG. Thanks. Can you keep one commit per module ?

Do you do the implementation also in the transport repo?

**Comment by jbaudoux** [comment]
@lmignon Is checking the toursolver failing test case. He fixed it in transport repo. I don't know why it is still complaining here

**Comment by jbaudoux** [comment]
> @lmignon Is checking the toursolver failing test case. He fixed it in transport repo. I don't know why it is still complaining here

fixed here https://github.com/OCA/wms/pull/928

**Review by lmignon — APPROVED** [review]
LGTM (Code review only)

**Comment by sebalix** [comment]
@jbaudoux there is one commit per module yes. I'll do the changes on `shipment_advice` yes

**Comment by sebalix** [comment]
https://github.com/OCA/stock-logistics-transport/pull/146

**Pull request merged**